pkg/api/pluginproxy/access_token_provider_test.go

```diff
@@ -34,6 +34,9 @@ func TestAccessToken(t *testing.T) {
 					"https://www.testapi.com/auth/monitoring.read",
 					"https://www.testapi.com/auth/cloudplatformprojects.readonly",
 				},
+				Headers: []plugins.AppPluginRouteHeader{
```

Comment:
I would like a test where headers is not set as well to verify the changes works without specifying any headers. Would suggest revert changes for this test case since you have another test case below verifying that headers are set as expected.